docs/security.md

```diff
@@ -15,48 +15,78 @@ Follow these instructions to [authenticate in strict mode](https://docs.mesosphe
 
-Both stores must be base64 encoded, for example:
 
-    cat keystore | base64 /u3+7QAAAAIAAAACAAAAAgA...
+The keystore and truststore are created using the [Java keytool][12]. The keystore
+must contain one private key and its signed public key. The truststore is optional
+and might contain a self-signed root-ca certificate that is explicitly trusted by Java.
+
+Both stores must be base64 encoded without newlines, for example:
+
+```bash
+cat keystore | base64 -w 0 > keystore.base64
+cat keystore.base64
+/u3+7QAAAAIAAAACAAAAAgA...
+```
 
 **Note:** The base64 string of the keystore will probably be much longer than the snippet above, spanning 50 lines or so.
 
-With this and the password `secret` for the keystore and the private key, your JSON options file will look like this:
+Add the stores to your secrets in the DC/OS secret store. For example, if your base64-encoded keystores 
+and truststores are server.jks.base64 and trust.jks.base64, respectively, then use the following 
+commands to add them to the secret store: 
+
+```bash
+dcos security secrets create /__dcos_base64__keystore --value-file server.jks.base64
+dcos security secrets create /__dcos_base64__truststore --value-file trust.jks.base64
+```
+
+You must add the following configurations to your `dcos spark run ` command.
+The ones in parentheses are optional:
 
-    {
-      "security": {
-        "ssl": {
-          "enabled": true,
-          "keyStoreBase64": "/u3+7QAAAAIAAAACAAAAAgA...”,
-          "keyStorePassword": "secret",
-          "keyPassword": "secret"
-        }
-      }
-    }
+```bash
 
-Install Spark with your custom configuration:
+dcos spark run --verbose --submit-args="\
+--keystore-secret-path=<path/to/keystore, e.g. __dcos_base64__keystore> \
+--keystore-password=<password to keystore> \
+--private-key-password=<password to private key in keystore> \
+(—-truststore-secret-path=<path/to/truststore, e.g. __dcos_base64__truststore> \)
+(--truststore-password=<password to truststore> \)
+(—-conf spark.ssl.enabledAlgorithms=<cipher, e.g., TLS_RSA_WITH_AES_128_CBC_SHA256> \)
+--class <Spark Main class> <Spark Application JAR> [application args]"
+```
 
-    dcos package install --options=options.json spark
+**Note:** If you have specified a space for your secrets other than the default value,
+`/spark`, then you must set `spark.mesos.task.labels=DCOS_SPACE:<dcos_space>`
+in the command above in order to access the secrets.
+See the [Secrets Documentation about spaces][13] for more details about spaces.
 
-Make sure to connect the DC/OS cluster only using an SSL connection (i.e. by using an `https://<dcos-url>`). Use the following command to set your DC/OS URL:
+**Note:** If you specify environment-based secrets with `spark.mesos.[driver|executor].secret.envkeys`,
+the keystore and truststore secrets will also show up as environment-based secrets,
+due to the way secrets are implemented. You can ignore these extra environment variables.
```

Comment:
This should also be mentioned in Limitations (https://docs.mesosphere.com/service-docs/spark/2.1.0-2.2.0-1/limitations/)

Reply:
Adding ...